### PR TITLE
Combine resource account references from multiple sources

### PR DESCRIPTION
**Pull request by mattwiller**

### CODE REVIEW

**Review by ianplunkett — APPROVED** [review]
I've tested this branch locally against the scenario described in this issue - https://github.com/medplum/medplum-customer-tasks/issues/227 and observed the desired results.  

This looks good to me.

File: packages/server/src/fhir/repo.ts [hunk 1]
```diff
@@ -1702,42 +1702,60 @@ export class Repository extends FhirRepository<PoolClient> implements Disposable
...
+    if (!existing && this.context.accessPolicy?.compartment?.reference) {
+      // If the creator's access policy specifies a compartment, then use it as the account.
+      // The writer's access policy is only applied at resource creation: simply editing a
+      // resource does NOT pull it into the user's account.
+      accounts.add(this.context.accessPolicy.compartment.reference);
     }
 
-    if (updated.resourceType !== 'Patient') {
+    if (updated.resourceType === 'Patient') {
+      // Otherwise, default to the existing value.
```
Comment (codyebberson) [inline packages/server/src/fhir/repo.ts:1725]: Minor nit, I think this comment is a copy/paste artifact.

```suggestion
      // When examining a Patient resource, we only look at the individual patient
      // We should not call `getPatients` and `readReference`
```

File: packages/server/src/fhir/repo.ts [hunk 2]
```diff
@@ -1702,42 +1702,60 @@ export class Repository extends FhirRepository<PoolClient> implements Disposable
...
+      if (existingAccounts?.length) {
+        for (const account of existingAccounts) {
+          accounts.add(account.reference as string);
+        }
+      }
+    } else {
       for (const patientRef of getPatients(updated)) {
         try {
           // If the resource is in a patient compartment, then lookup the patient.
           const patient = await getSystemRepo().readReference(patientRef);
```
Comment (codyebberson) [inline packages/server/src/fhir/repo.ts:1736]: Minor nit, we could now use `readReferences` to make this a batch call.  Low priority, that can be a future optimization.

**Review by codyebberson — APPROVED** [review]
Some optional minor nits

**Pull request merged**